src/bitcoin.cpp

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) 2024 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <bitcoin-build-config.h> // IWYU pragma: keep
+
+#include <clientversion.h>
+#include <util/fs.h>
+#include <util/exec.h>
+#include <util/strencodings.h>
+#include <util/translation.h>
+
+#include <iostream>
+#include <string>
+#include <tinyformat.h>
+#include <vector>
+
+const TranslateFn G_TRANSLATION_FUN{nullptr};
+
+static constexpr auto HELP_USAGE = R"(Usage: %s [OPTIONS] COMMAND...
+
+Options:
+  -m, --multiprocess     Run multiprocess binaries bitcoin-node, bitcoin-gui.
+  -M, --monolithic       Run monolithic binaries bitcoind, bitcoin-qt. (Default behavior)
+  -v, --version          Show version information
+  -h, --help             Show this help message
+
+Commands:
+  gui [ARGS]     Start GUI, equivalent to running 'bitcoin-qt [ARGS]' or 'bitcoin-gui [ARGS]'.
+  daemon [ARGS]  Start daemon, equivalent to running 'bitcoind [ARGS]' or 'bitcoin-node [ARGS]'.
```

Comment:
On `master`, when `-daemon` is used to start `bitcoind`, it means logs won't be printed to the console. I believe that's also the literal meaning of  daemon: https://en.wikipedia.org/wiki/Daemon\_(computing

This PR changes makes the keyword ambiguous to mean bitcoin node executable and start the executable in the background so doing the two things at once is now a bit awkard  `bitcoin daemon -daemon`.

Should the  be `node`?

So that we can distinguish the two and have a cleaner command `bitcoin node -daemon`.

Reply:
TIL we have a `-daemon` option. Indeed we'll have to pick a different term here.

Reply:
I think it is fine as is to be honest. `bitcoind -daemon` already means the same thing (albeit abbreviated).

Reply:
I think `-daemon` detaches, so you can't ctrl + c it.

Reply:
> `bitcoin daemon -daemon`

Good point. Indeed I find that super not-intuitive, I did not think about that before. I would strongly prefer to rename it to `node`, e.g. `bitcoin node -daemon`.

> `bitcoind -daemon` already means the same thing

Eh, but somehow I don't find it so perplexing. Or even if it is as confusing, now we have a chance to pick better names.

Reply:
Good catch! I do think daemon could be ok but I agree node is better. I will update this (and I also want to look into @theStack's suggestion to use subprocess.h, which I was unaware of and could simplify the PR)